### DISCUSSION TITLE
ArgoCD failing to sync resource with metav1.Duration

### DISCUSSION THREAD

**Discussion by vrv501** [argoproj/argo-cd#14229] [Q&A]
Hi I'm trying to deploy cert-manager.io/Certificate resource through argoCD. The manifest for this resource has the below field

```
spec:
  duration: 4567h
```

After the resource is applied to cluster by argoCD, ClusterIssuer mutates the above resource resulting in

```
spec:
  duration: 4567h0m0s
```

After looking through clusterIssuer code, it looks like its unmarshalling above field as `metav1.Duration` due to which mutation happens.

Due to this the argoApplication is always OutOfSync. Can I configure argoCD to map `spec.Duration` to metav1.Duration(kubernetes built-in type) 

I couldn't find any known kubernetes built-in types which match metav1.Duration atleast from the well_known types doc that is linked in argoCD docs. 
I do not want to add this to ignoreDifferences

**Comment by vrv501** [comment]
@crenshaw-dev Appreciate if you can help with few pointers around this 🙏

**Comment by davidmontoyago** [comment]
hi @vrv501, my team also experienced this issue recently while syncing cert-manager's `Certificates`. i can confirm the discrepancy also appears with `kubecfg diff`, which confirms the quirk/bug on cert-manager's itself -- in our case, we opted for adding the long format (`4567h0m0s`) on the source manifests which quickly resolved the out-of-sync. in think this is one of the cases for which `ignoreDifferences` was created.

**Comment by david-igou** [comment]
I encountered this issue when working with a helm chart I could not edit. I worked around with `ignoreDifferences`:

```yaml
spec:
  ignoreDifferences:
  - group: cert-manager.io
    kind: Certificate
    jsonPointers:
    - /spec/duration
```
